drivers/ethernet/eth_stm32_hal.c

```diff
@@ -830,11 +833,11 @@ static int eth_stm32_hal_set_config(const struct device *dev,
 	struct eth_stm32_hal_dev_data *dev_data;
```

Comment:
```suggestion
	int ret = -ENOTSUP;
	struct eth_stm32_hal_dev_data *dev_data;
```

You will need it later.